Observer/Config/Save.php

```diff
@@ -98,40 +107,44 @@ public function execute(Observer $observer)
-                return true;
-            }
+            if ($this->isYotpoSettingsChanged($changedPaths)) {
+                $this->cacheTypeList->cleanType(Config::TYPE_IDENTIFIER);
+                $this->appConfig->reinit();
+                $scope = $scopeDetails['scope'];
+                $scopes = $scopeDetails['scopes'];
+                $appKey = $this->yotpoConfig->getAppKey($scopeId, $scope);
+
+                if ($scope !== ScopeInterface::SCOPE_STORE && !$this->yotpoConfig->isSingleStoreMode()) {
+                    $this->resetStoreCredentials($scopeId, $scopes);
+                    return true;
+                }
 
-            //Check if appKey is unique:
-            if ($appKey) {
-                foreach ((array) $this->yotpoConfig->getAllStoreIds() as $key => $storeId) {
-                    if (($scopeId && $storeId != $scopeId) && $this->yotpoConfig->getAppKey($storeId) === $appKey) {
-                        $this->resetStoreCredentials($scopeId, $scopes);
-                        throw new AlreadyExistsException(__(
-                            "The APP KEY you've entered is already in use by another store on this system.
-                            Note that Yotpo requires a unique set of APP KEY & SECRET for each store."
-                        ));
+                //Check if appKey is unique:
+                if ($appKey) {
+                    foreach ((array) $this->yotpoConfig->getAllStoreIds() as $key => $storeId) {
+                        if (($scopeId && $storeId != $scopeId) && $this->yotpoConfig->getAppKey($storeId) === $appKey) {
+                            $this->resetStoreCredentials($scopeId, $scopes);
+                            throw new AlreadyExistsException(__(
+                                "The APP KEY you've entered is already in use by another store on this system.
+                                Note that Yotpo requires a unique set of APP KEY & SECRET for each store."
+                            ));
+                        }
                     }
                 }
-            }
 
-            if ($this->yotpoConfig->isEnabled($scopeId, $scope) &&
-                !($this->yotpoApi->createAuthToken($scopeId, $scope))) {
-                $this->resetStoreCredentials($scopeId, $scopes);
-                throw new AlreadyExistsException(__(
-                    "Please make sure the APP KEY and SECRET you've entered are correct"
-                ));
+                if ($this->yotpoConfig->isEnabled($scopeId, $scope) &&
+                    !($this->yotpoApi->createAuthToken($scopeId, $scope))) {
+                    $this->resetStoreCredentials($scopeId, $scopes);
+                    throw new AlreadyExistsException(__(
+                        "Please make sure the APP KEY and SECRET you've entered are correct"
+                    ));
+                }
+            }
+            if ($scopeId && $this->isYotpoKeysChanged($changedPaths)) {
```

Comment:
We don't need it because the Yotpo Enablement, App key are available only under the store scope.  So isYotpoSettingsChanged() and isYotpoAppKeyChanged() methods would return false if it is not in a store's scope